stl/inc/algorithm

```diff
@@ -2841,6 +2841,61 @@ _FwdIt2 replace_copy_if(_ExPo&&, _FwdIt1 _First, _FwdIt1 _Last, _FwdIt2 _Dest, _
 }
 #endif // _HAS_CXX17
 
+#ifdef __cpp_lib_concepts
+namespace ranges {
+    // VARIABLE ranges::generate
+    class _Generate_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        // clang-format off
+        template <input_or_output_iterator _Out, sentinel_for<_Out> _Se, copy_constructible _Fn>
+            requires invocable<_Fn&> && indirectly_writable<_Out, invoke_result_t<_Fn&>>
+        constexpr _Out operator()(_Out _First, _Se _Last, _Fn _Gen) const {
+            _Adl_verify_range(_First, _Last);
+            auto _UFirst      = _Get_unwrapped(_STD move(_First));
+            const auto _ULast = _Get_unwrapped(_STD move(_Last));
+            for (; _UFirst != _ULast; ++_UFirst) {
+                *_UFirst = _Gen();
+            }
+
+            _Seek_wrapped(_First, _STD move(_UFirst));
+            return _First;
+        }
+
+        template <class _Rng, copy_constructible _Fn>
+            requires invocable<_Fn&> && output_range<_Rng, invoke_result_t<_Fn&>>
+        constexpr borrowed_iterator_t<_Rng> operator()(_Rng&& _Range, _Fn _Gen) const {
+            return (*this)(_RANGES begin(_Range), _RANGES end(_Range), _Gen);

```

Comment:
I believe this should use `_Pass_fn`, which is an internal helper to reduce the expense of passing function objects within STL machinery. (In the classic C++98/03 era, function objects were assumed to be cheap to copy, but that is less true these days given the runtime expense of `std::function` and similar machinery.) I observe that every other `_Rng&& _Range` overload uses `_Pass_fn` except for `for_each` which is special (it uses `_STD move` because its function object must ultimately be returned).

```suggestion
            return (*this)(_RANGES begin(_Range), _RANGES end(_Range), _Pass_fn(_Gen));
```